core/io/pck_reader.h

```diff
@@ -0,0 +1,57 @@
+/**************************************************************************/
+/*  pck_reader.h                                                          */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef PCK_READER_H
+#define PCK_READER_H
+
+#include "core/io/file_access_pack.h"
+#include "core/object/ref_counted.h"
+
+class PCKReader : public RefCounted {
+	GDCLASS(PCKReader, RefCounted)
+
+	PCKFile file;
+
+protected:
+	static void _bind_methods();
+
+public:
+	Error open(String p_path, uint64_t offset);
```

Comment:
```suggestion
	Error open(const String &p_path, uint64_t p_offset);
```
Pass by reference and prefix arguments with `p_`

Reply:
Interestingly enough the `ZIPReader` is not using pass by reference either.

I'll create a pull request to fix it once I find the time (in the next couple of days)

Reply:
It's not necessarily significant to change it there, the difference is relatively minor, but it is still cheaper to pass by explicit reference (`String` is ref-counted so all it does to pass by copy is adding a reference, which does have a small but not entirely zero cost, with some atomics involved)